### PR TITLE
feat: support using http api upload license

### PR DESCRIPTION
**Pull request by Gala-R**

### CODE REVIEW

File: sidecar/reloader/main.go [hunk 1]
```diff
@@ -106,47 +112,71 @@ func main() {
...
+							dir, fileName := filepath.Split(event.Name)
+							if dir == fmt.Sprintf("%s/", pluginConfigDir) {
+								pluginName := strings.TrimSuffix(fileName, ".conf")
+								/*
+									curl -i --basic -u admin:public -X PUT
+									"http://localhost:8081/api/v4/nodes/emqx@127.0.0.1/plugins/emqx_delayed_publish/reload"
+								*/
+								loadUrl.Path = fmt.Sprintf("/api/v4/plugins/%s/reload", pluginName)
+								requestWebhook(loadUrl, "PUT", nil, username, password)
+							} else if dir == licenseFileDir {
```
Comment (Rory-Z) [inline sidecar/reloader/main.go:132]: How about like this: 
```
if event.Name == licenseFilePath {
// update license
} else {
// reload plugins
}
```

and `licenseFileDir` can be delete

**Comment by Rory-Z** [comment]
We need to find a way to cover this part of the code with test cases

File: sidecar/reloader/main.go [hunk 2]
```diff
@@ -188,7 +218,5 @@ func requestWebhook(url url.URL, username, password string) {
 		log.Println("ioutil read error:", err)
 		return
 	}
-	log.Println("http response url:", url.String())
-	log.Println("status:", resp.Status)
-	log.Println("body:", string(body))
+	log.Printf("http response \n { url: %s, status: %s, body: %s }\n", url.String(), resp.Status, string(body))
```
[the remark this answers is not in the document]
Comment (Gala-R) [reply sidecar/reloader/main.go:221]: why print three logs ?

File: sidecar/reloader/main.go (hunk 1 above)
Comment (Gala-R) [reply sidecar/reloader/main.go:137]: no matter, th next version of the api is directly bound to the file

File: controllers/apps/v1beta3/emqx_handler.go [hunk 3]
```diff
@@ -990,3 +990,6 @@ func updateLicenseForsts(sts *appsv1.StatefulSet, license *corev1.Secret) *appsv
 		},
 	)
 }
+
+
+
```
[the remark this answers is not in the document]
Comment (Gala-R) [reply controllers/apps/v1beta3/emqx_handler.go]: Done

File: sidecar/reloader/main.go [hunk 4]
```diff
@@ -106,47 +112,71 @@ func main() {
...
+				log.Fatal(err)
+			}
+			fileCheck[fullFile] = getMD5(fullFile)
 		}
-		fullFile := filepath.Join(configFilePath, file)
-		log.Printf("watching config file: %q", fullFile)
+	}
 
-		err = watcher.Add(fullFile)
+	// 	license file
```
[the remark this answers is not in the document]
Comment (Gala-R) [reply sidecar/reloader/main.go]: Done

File: sidecar/reloader/main.go [hunk 5]
```diff
@@ -106,47 +112,71 @@ func main() {
...
+								bytes, err := os.ReadFile(event.Name)
+								if err != nil {
+									log.Fatal(err)
+								}
+								/*
+									curl -X POST --basic -u admin:public -d @<(jq -sR '{license: .}' < path/to/new.license)
+									'http://localhost:8081/api/v4/license/upload
+								*/
+								loadUrl.Path = "/api/v4/license/upload"
+								body := fmt.Sprintf("{license:%s}", string(bytes))
```
[the remark this answers is not in the document]
Comment (Gala-R) [reply sidecar/reloader/main.go]: Done

File: sidecar/reloader/main.go [hunk 6]
```diff
@@ -106,47 +112,71 @@ func main() {
...
-							url := url.URL{
-								Scheme: "http",
-								Host:   fmt.Sprintf("%s:%d", host, port),
-								Path:   fmt.Sprintf("/api/v4/plugins/%s/reload", name),
+							dir, fileName := filepath.Split(event.Name)
+							if dir == fmt.Sprintf("%s/", pluginConfigDir) {
+								pluginName := strings.TrimSuffix(fileName, ".conf")
+								/*
+									curl -i --basic -u admin:public -X PUT
+									"http://localhost:8081/api/v4/nodes/emqx@127.0.0.1/plugins/emqx_delayed_publish/reload"
```
[the remark this answers is not in the document]
Comment (Gala-R) [reply sidecar/reloader/main.go]: Done

File: sidecar/reloader/main.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (Rory-Z) [reply sidecar/reloader/main.go]: You also printed three lines of log

File: sidecar/reloader/main.go [hunk 7]
```diff
@@ -106,47 +116,75 @@ func main() {
...
+								if err != nil {
+									log.Fatal(err)
+								}
+								/*
+									curl -X POST --basic -u admin:public -d @<(jq -sR '{license: .}' < path/to/new.license)
+									"http://localhost:8081/api/v4/license/upload"
+								*/
+								loadUrl.Path = "/api/v4/license/upload"
+								license := License{License: bytes}
+								body, err := json.Marshal(license)
```
Comment (Rory-Z) [inline sidecar/reloader/main.go:139]:
```suggestion
								license := map[string][]byte{"license": bytes}
								body, err := json.Marshal(license)
```

**Review by Rory-Z — COMMENTED** [review]
No need license struct

File: sidecar/reloader/main.go (hunk 7 above)
Comment (Gala-R) [reply sidecar/reloader/main.go:139]: Done

File: sidecar/reloader/main.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (Gala-R) [reply sidecar/reloader/main.go]: reset

**Review by Rory-Z — APPROVED** [review]

**Pull request merged**